pkg/controller/cloudprivateipconfig/cloudprivateipconfig_controller.go

```diff
@@ -179,24 +179,98 @@ func (c *CloudPrivateIPConfigController) SyncHandler(key string) error {
 		return err
 	}
 
-	// At most one of nodeToAdd or nodeToDel will be set
-	nodeToAdd, nodeToDel := c.computeOp(cloudPrivateIPConfig)
+	// At most one of nodeNameToAdd or nodeNameToDel will be set
+	nodeNameToAdd, nodeNameToDel := c.computeOp(cloudPrivateIPConfig)
 	switch {
 	// Dequeue on NOOP, there's nothing to do
-	case nodeToAdd == "" && nodeToDel == "":
+	case nodeNameToAdd == "" && nodeNameToDel == "":
 		return nil
-	case nodeToDel != "":
-		klog.Infof("CloudPrivateIPConfig: %q will be deleted from node: %q", key, nodeToDel)
+	case nodeNameToAdd != "" && nodeNameToDel != "":
+		klog.Infof("CloudPrivateIPConfig: %q will be moved from node %q to node %q", key, nodeNameToDel, nodeNameToAdd)
+		nodeToDel, err := c.nodesLister.Get(nodeNameToDel)
+		if err != nil {
+			return err
+		}
+		nodeToAdd, err := c.nodesLister.Get(nodeNameToAdd)
+		if err != nil {
+			return err
+		}
+
+		status = &cloudnetworkv1.CloudPrivateIPConfigStatus{
+			Node: nodeNameToDel,
+			Conditions: []metav1.Condition{
+				{
+					Type:               string(cloudnetworkv1.Assigned),
+					Status:             metav1.ConditionUnknown,
+					ObservedGeneration: cloudPrivateIPConfig.Generation,
+					LastTransitionTime: metav1.Now(),
+					Reason:             cloudResponseReasonPending,
+					Message:            "Moving IP address",
+				},
+			},
+		}
+		if cloudPrivateIPConfig, err = c.updateCloudPrivateIPConfigStatus(cloudPrivateIPConfig, status); err != nil {
+			return fmt.Errorf("error updating CloudPrivateIPConfig: %q during move operation, err: %v", key, err)
+		}
 
-		node, err := c.nodesLister.Get(nodeToDel)
+		// This is a blocking call. If the IP is not assigned then don't treat
+		// it as an error.
+		var i interface{} = c.cloudProviderClient
+		withMover, ok := i.(cloudprovider.CloudProviderWithMoveIntf)
+		if !ok {
+			return fmt.Errorf("cannot convert driver to the interface with move abilities, this should never happen")
+		}
+		if moveErr := withMover.MovePrivateIP(ip, nodeToAdd, nodeToDel); moveErr != nil && !errors.Is(moveErr, cloudprovider.NonExistingIPError) {
```

Comment:
extreme nit (also for other lines): I like keeping lines at 120 chars max + a few if I really cannot avoid it (which may be the case here). I know that's not the case for the older code in the repo and we don't really have a standard, but it's easier to read IMO. Feel free to ignore this and keep as is

Reply:
Looking at the code, I think ReleasePrivateIP is the only function atm that can throw the NonExistingIPError (which is logical, in case of a Release, we have a means of reporting the issue, but we ignore it). I'm not sure if it makes sense for MovePrivateIP to throw such an error. Currently as far as I see it doesn't, and so we don't have to deal with it hear, either?